restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +90,20 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  public static void getAbsentFieldsDefaultValues(RecordDataSchema dataSchema, DataMap dataMap)
+  {
+    for (RecordDataSchema.Field field : dataSchema.getFields())
+    {
+      if (!dataMap.containsKey(field.getName()))
+      {
+        if (field.getDefault() != null)
+        {
+          dataMap.put(field.getName(), field.getDefault());
```

Comment:
as discussed update this logic to fill nested defaults if field value is not null